CodeGeneration/Sources/SyntaxSupport/Child.swift

```diff
@@ -103,6 +103,11 @@ public class Child {
     return .identifier(lowercaseFirstWord(name: name))
   }
 
+  /// A type of this child.
+  public var type: TypeSyntax {
+    return "\(raw: name.withFirstCharacterUppercased)"
+  }
```

Comment:
Sorry, after reading this PR again, I noticed that this isn’t actually the type of the child, unless it has multiple node choices. What this technically is, is the name that a nested `SyntaxChoices` type gets, if it gets created. With this in mind, a better name would be `syntaxChoicesType` and it should fail if the child doesn’t have `kind == .nodeChoices` to prevent misuse.

```suggestion
  /// If this child has node choices, the type that the nested `SyntaxChildChoices` type should get.
  /// 
  /// For any other kind of child nodes, accessing this property crashes.
  public var syntaxChoicesType: TypeSyntax {
    precondition(kind.isNodeChoices, "Cannot get syntaxChoicesType for node that doesn’t have nodeChoices")
    return "\(raw: name.withFirstCharacterUppercased)"
  }
```

And I think most of the uses of `type` in `ValidateSyntaxNodes` should be `varOrCaseName` or, where necessary `varOrCaseName.description`.

Reply:
It seems very reasonable. I just wondering what about places like this one: https://github.com/apple/swift-syntax/blob/83348e7eb826b978896421b2a233a4290596eb58/CodeGeneration/Sources/generate-swiftsyntax/templates/swiftparser/ParserTokenSpecSetFile.swift#L23C4-L29 ? Do we wanna use `name.withFirstCharacterUppercased` instead of `syntaxChoicesType`?

Reply:
@ahoppen  I'm sorry, but I got lost. Could you please summarize once more what we want to achieve?

Reply:
Did you mean introducing two new properties on `Child`: one for `kind.isNodeChoices` and the other for `kind.isToken`?

Reply:
Yes, exactly.

Reply:
Done